ochopod/frameworks/marathon.py

```diff
@@ -329,6 +331,28 @@ def _shell():
                 return json.dumps(out), 200, {'Content-Type': 'application/json; charset=utf-8'}
 
             #
+            # - arbitrary tool invokation (used for users to define custom behavoir)
+            #
+            @web.route('/tools', methods=['POST'])
+            def _tools():
+                logger.info('http in -> /tools')
+
+                def _help():
+                    return ['available commands -> %s' % ', '.join(sorted([tool.tag for tool in tools]))] if tools else []
+
+                xtools = request.headers['X-Tools']
+                if xtools == 'help':
+                    return json.dumps(_help()), 200, {'Content-Type': 'application/json; charset=utf-8'}
+
+                result = []
+                for tool in tools:
+                        if xtools.startswith(tool.tag):
```

Comment:
mh why this statement ? can you specify more than one tool in the X-Tools header ?